### ISSUE TITLE
array use bug

### ISSUE THREAD

**Issue by rogertheriault** [Automattic/VIP-Coding-Standards#223]
the bot frequently warns about an undefined variable when accessing array attributes

```
$cat_json_data = [];
		$cat_json_data[] = [
  @wpcomvip-vipgoci-bot
wpcomvip-vipgoci-bot 3 hours ago
❗️ Warning: Variable $cat_json_data is undefined.
```

and similarly in a foreach loop these false warnings have been seen

**Comment by GaryJones** [comment]
@rogertheriault Can you please give some more context to this - more code?

**Comment by GaryJones** [comment]
The warning is coming from either [here](https://github.com/Automattic/VIP-Coding-Standards/blob/0363baf7ae0d92b2662bc0140da12709cbc7ee02/WordPressVIPMinimum/Sniffs/Variables/VariableAnalysisSniff.php#L493-L498) or [here](https://github.com/Automattic/VIP-Coding-Standards/blob/0363baf7ae0d92b2662bc0140da12709cbc7ee02/WordPressVIPMinimum/Sniffs/Variables/VariableAnalysisSniff.php#L729-L733).

Relevant: #236

@gudmdharalds How much flexibility does the bot have in ignoring certain checks / sniffs?

**Comment by gudmdharalds** [comment]
@GaryJones: Relatively high. If we want, we can specify that certain sniffs are to be ignored. That should not be complicated to implement.

**Comment by GaryJones** [comment]
See #236.

**Comment by GaryJones** [comment]
Now that #449 is completed to use VariableAnalysis proper, I've confirmed that the provided example above doesn't give an error with the `develop` version of `WordPress-VIP-Go`, so this is now fixed for VIPCS 2.2.0.
